docs/_includes/versions-dropdown.html

```diff
@@ -9,9 +9,20 @@
   {%- endif %}">
 {%- for item in versions reversed %}
 {%- unless item[1].hidden and spec_version != item[0] %}
-  {% assign page_url_parts = page.url | split: '/' %}
-  {% assign page_url = page_url_parts | join: '/' %}
-  <option value="{{ page_url | replace: spec_version, item[0] | relative_url }}" {% if spec_version == item[0] %}selected{% endif %} class="inline-block">{{item[1].name}}</option>
+  {%- assign page_url_parts = page.url | split: '/' %}
+  {%- assign page_url = page_url_parts | join: '/' %}
+  {%- assign other_version = page_url | replace: spec_version, item[0] | relative_url %}
+  {% comment %}
+    If the other version of the page doesn't exist disable the option
+  {% endcomment %}
+  {%- assign p = site.pages | where_exp: "page", "page.url contains other_version" | first %}
+  {%- if p %}
+    {%- assign disabled = "" %}
+  {%- else %}
+    {%- assign disabled = "disabled " %}
+  {%- endif %}
+<option {{ disabled }} value="{{ other_version }}" {% if spec_version == item[0] %}selected{% endif %} class="inline-block">{{item[1].name}}</option>
```

Comment:
I like your change! It made it much easier to read.

Here's a suggestion to go one step further, since selected and disabled are mutually exclusive. (The `p and` is probably unnecessary, but it probably makes it more clear.)

```suggestion
  {%- if p and spec_version == item[0] %}
    {%- assign state = "selected" %}
  {%- elsif p %}
    {%- assign state = "" %}  
  {%- else %}
    {%- assign state = "disabled" %}
  {%- endif %}
<option {{ state }} value="{{ other_version }}" class="inline-block">{{item[1].name}}</option>
```

Reply:
Yes, that's even nicer. Thanks!